presto-spi/src/main/java/io/prestosql/spi/block/SingleMapBlock.java

```diff
@@ -154,7 +147,7 @@ public int seekKey(Object nativeValue)
 
         long hashCode;
         try {
-            hashCode = (long) keyNativeHashCode.invoke(nativeValue);
+            hashCode = (long) mapType.getKeyNativeHashCode().invoke(nativeValue);
```

Comment:
These invokes seem to repeated throughout the class. Maybe introduce some helper methods.

Reply:
Tried that.  There are a bunch of subtle unique invocation shapes here.